credentials/xds/xds.go

```diff
@@ -129,6 +152,14 @@ func (hi *HandshakeInfo) SetAcceptedSANs(sans []string) {
 	hi.mu.Unlock()
 }
 
+// SetRequireClientCert updates whether a client cert is required during the
+// ServerHandshake(). A value of true indicates that we are performing mTLS.
+func (hi *HandshakeInfo) SetRequireClientCert(require bool) {
```

Comment:
Just curious: what is the motivation behind this change? Does xDS allows one-side TLS and mTLS connections in the same time?

Reply:
Also, it seems this function is only used in tests. Do we expect to expose it as a public function?

Reply:
I do not quite understand what you mean by "Does xDS allows one-side TLS and mTLS connections in the same time?". In a single gRPC application, one could be exposing multiple services. And this can be done with a single `xds.GRPCServer` and single `xds.Credentials`. But each of these services could listen on different ports. This means that the listener configuration could be different for them, and hence could be TLS for one and mTLS for another. And since the credentials object is shared between the two services, we need to pass all handshake info on a per-connection basis and we cannot store anything at the credentials level.

This function is not something that will only be used by tests. This will be used by `xds.GRPCServer` to set this field based on what it receives from the control plane.

Reply:
OK I see. I think your explanation explains it well :)